redos/7/Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM centos:7 as prepare
+MAINTAINER Sergei Vorontsov "piligrim@rootnix.net"
+
+ENV REPO_URL=http://mirror.yandex.ru/redos/7.3/x86_64/os/ \
+    CHROOT_DIR="/chroot"
+
+RUN mkdir -p $CHROOT_DIR
+
+RUN for rpm in `curl -s $REPO_URL | grep 'redos-release' | sed 's/<.*>\(.*\)<\/.*>/\1/' | awk '{print $1}'`; do if [[ $rpm =~ redos-release-[0-9].[0-9]-([0-9]|[0-9][0-9]).el7.noarch.rpm ]]; then curl -o $CHROOT_DIR/redos-release.rpm $REPO_URL$rpm; fi; done
+
+#setup base system
+RUN rpm --root=$CHROOT_DIR --rebuilddb
+RUN rpm --root=$CHROOT_DIR --nodeps --quiet -i $CHROOT_DIR/redos-release.rpm
+RUN cp $CHROOT_DIR/etc/pki/rpm-gpg/RPM-GPG-KEY-RED-SOFT /etc/pki/rpm-gpg/RPM-GPG-KEY-RED-SOFT
+
+RUN yum -y --installroot=$CHROOT_DIR update
+
+RUN yum -y --installroot=$CHROOT_DIR install -y \
```

Comment:
I guess that some of the packages are from CentOS 7 itself (at least I see no `zypper` on the Yandex's RedOS mirror). Or at least they can be installed here. Is it deliberately? If so, please, add a comment, which will answer the following questions:

* Is it okay to place packages from CentOS 7 here (not from RedOS)?
* If so, how we should limit it to don't end up with CentOS 7 under RedOS name?
* If there are any other caveats with packages selection.

If only RedOS 7.3 packages should be allowed here, we should let the code fail if a package not from RedOS 7.3 is selected. I guess we can remove all mirrors except RedOS's one. If it is not so, I would like to see an explanation regarding it either in the code or in the commit message.

----

Update: I run the build and see the following warnings:

```
No package devtoolset available.
Package yum-utils is obsoleted by dnf-utils, trying to install dnf-utils-4.0.18-3.el7.noarch instead
No package zypper available.
```

So I guess the following:

* Packages from CentOS 7 are not installed here (and I don't know how it works, TBH; `--installroot` uses only chroot's mirrors?)
* Non-existing selections are ignored (it would be good to fail in this case).
* `zypper` and `devtoolset` actually not installed.
* It worth to install dnf-utils instead of yum-utils.

Despite that everything looks mostly good I would like to see some explanations (I kept my original message above, so you can see what was confusing for me).

Reply:
I replaced yum-utils with dnf-utils and removed devtoolset.

Reply:
It seems, it was not done, but I did.